src/Modules/IexecPoco.ts

```diff
@@ -1,461 +1,432 @@
-  deal.workerpoolPrice = toRLC(viewedDeal.workerpool.price);
-  deal.trust = viewedDeal.trust;
-  deal.category = viewedDeal.category.toString();
-  deal.tag = viewedDeal.tag;
-  deal.requester = viewedDeal.requester.toHex();
-  deal.beneficiary = viewedDeal.beneficiary.toHex();
-  deal.callback = viewedDeal.callback.toHex();
-  deal.params = viewedDeal.params;
-  deal.startTime = viewedDeal.startTime;
-  deal.botFirst = viewedDeal.botFirst;
-  deal.botSize = viewedDeal.botSize;
-  deal.workerStake = viewedDeal.workerStake;
-  deal.schedulerRewardRatio = viewedDeal.schedulerRewardRatio;
-  deal.apporder = event.params.appHash.toHex();
-  deal.datasetorder = event.params.datasetHash.toHex();
-  deal.workerpoolorder = event.params.workerpoolHash.toHex();
-  deal.requestorder = event.params.requestHash.toHex();
-  deal.timestamp = event.block.timestamp;
-  deal.save();
-
-  const dataset = deal.dataset;
-
-  let apporder = fetchApporder(event.params.appHash.toHex());
-  apporder.app = deal.app;
-  apporder.appprice = deal.appPrice;
-  apporder.save();
-
-  let datasetorder = fetchDatasetorder(event.params.datasetHash.toHex());
-  if (dataset) datasetorder.dataset = dataset;
-  datasetorder.datasetprice = deal.datasetPrice;
-  datasetorder.save();
-
-  let workerpoolorder = fetchWorkerpoolorder(
-    event.params.workerpoolHash.toHex()
-  );
-  workerpoolorder.workerpool = deal.workerpool;
-  workerpoolorder.workerpoolprice = deal.workerpoolPrice;
-  workerpoolorder.save();
-
-  let requestorder = fetchRequestorder(event.params.requestHash.toHex());
-  requestorder.requester = viewedDeal.requester.toHex();
-  requestorder.save();
-
-  let orderMatchedEvent = new OrdersMatched(createEventID(event));
-  orderMatchedEvent.transaction = logTransaction(event).id;
-  orderMatchedEvent.timestamp = event.block.timestamp;
-  orderMatchedEvent.deal = event.params.dealid.toHex();
-  orderMatchedEvent.save();
-
-  let protocol = fetchProtocol();
-  protocol.dealsCount = protocol.dealsCount.plus(BigInt.fromI32(1));
-  protocol.tasksCount = protocol.tasksCount.plus(deal.botSize);
-  protocol.save();
+    let contract = IexecInterfaceTokenContract.bind(event.address);
+    let viewedDeal = contract.viewDeal(event.params.dealid);
+    // The `sponsor` has been introduced on Bellecour for the PoCo v5.5.0 release:
+    // https://blockscout.bellecour.iex.ec/tx/0x71b904f526a9be218d35748f57d74ef6da20d12c88f94cfa1ec5ae2de187cb98
+    // TODO: Use grafting instead, see https://thegraph.com/docs/en/subgraphs/cookbook/grafting/
+    const sponsor =
+        chainName == 'bellecour' && event.block.number < BigInt.fromI32(30277938)
+            ? Address.zero().toHexString()
+            : viewedDeal.sponsor.toHex();
+    fetchAccount(viewedDeal.requester.toHex()).save();
+    fetchAccount(viewedDeal.beneficiary.toHex()).save();
+    fetchAccount(viewedDeal.callback.toHex()).save();
+    fetchAccount(sponsor).save();
```

Comment:
Shouldn't this run only if sponsor != `0x0`?

Reply:
To go further, I'm not even sure this line is generally required, but decided to stick with current harmonization, just like `callback` for instance -which could be empty -, what do you think?